pkg/reconciler/managed/reconciler.go

```diff
@@ -306,6 +308,15 @@ type ExternalObservation struct {
 
 // An ExternalCreation is the result of the creation of an external resource.
 type ExternalCreation struct {
+	// ResourceLateInitialized should be true if the managed resource's spec was
+	// updated during its creation. A Crossplane provider may update a
```

Comment:
nitpick: this includes annotations (e.g. external-name) and potentially other things outside of the spec right?